### ISSUE TITLE
[a11y]: SideNavDivider -  role 'separator' is not valid for the element <li>

### ISSUE THREAD

**Issue by tomleah** [carbon-design-system/carbon#14479]
### Package

carbon-components-react

### Browser

Chrome

### Operating System

Windows

### Package version

v7.59.10

### React version

v16.13.1

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker - Latest Deployment

### Description

Using the <SideNavDivider/> within the shell sidebar results in a accessibility violation "The ARIA role 'separator' is not valid for the element \<li>"

### WCAG 2.1 Violation

4.1.2 Name, Role, Value

### Reproduction/example

https://stackblitz.com/edit/github-cqny7t?file=src%2FApp.jsx

### Steps to reproduce

Use a SideNavDivider within SideNavItems and run accessibility checker

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by joshualiu** [comment]
Noticed the same issue on `@carbon/react@1.38.0`
<img width="751" alt="Screenshot 2023-09-27 at 10 03 31" src="https://github.com/carbon-design-system/carbon/assets/25513105/4f44fee3-0bcb-4cf1-8af1-55bea2cffa9a">
